File: URLShortener.Front/src/app/services/interfaces/IAccountService.ts
import { HttpHeaders } from "@angular/common/http";
import { Observable } from "rxjs";
import { AccountModel } from "src/app/models/AccountModel";
import { SignInModel } from "src/app/models/SignInModel";
import { SignUpModel } from "src/app/models/SignUpModel";
import { UpdateAccountModel } from "src/app/models/UpdateAccountModel";
import { IBaseResponse } from "../BaseResponse/IBaseResponse";

export interface IAccountService<T>{
    delete(id: string): Observable<IBaseResponse<T>>;
    update(id: string, entity: UpdateAccountModel): Observable<IBaseResponse<T>>;
    get(id: string): Observable<IBaseResponse<T>>;
    getCurrentAccount(): T;
    makeJWTHeader(): HttpHeaders;
    signUp(model: SignUpModel): Observable<IBaseResponse<T>>;
    signIn(model: SignInModel): Observable<IBaseResponse<T>>;
    refreshCurrentAccount(): void;
    saveCurrentAccount(entity: T): void;
    signOut(): void;
}

File: URLShortener.Back/URLShortener.Services/Interfaces/IAccountService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using URLShortener.DataAccessLayer.BaseResponse;
using URLShortener.DataAccessLayer.Entities;
using URLShortener.Models;

namespace URLShortener.Services.Interfaces
{
    public interface IAccountService
    {
        public Task<IBaseResponse<AccountModel>> DeleteAccount(string id);
        public Task<IBaseResponse<AccountModel>> UpdateAccount(string id, UpdateAccountModel model);
        public Task<IBaseResponse<AccountModel>> GetAccountById(string id);

        public Task<IBaseResponse<AccountModel>> RefreshAccountJWT(string accountId);
        public Task<IBaseResponse<AccountModel>> GetAccountByUsername(string username);
        public Task<IBaseResponse<AccountModel>> SignUp(SignUpModel model);
        public Task<IBaseResponse<AccountModel>> SignIn(SignInModel model);
    }
}
